packages/runtime-core/src/hydration.ts

```diff
@@ -938,10 +938,10 @@ function resolveCssVars(
   ) {
     const cssVars = instance.getCssVars()
     for (const key in cssVars) {
-      expectedMap.set(
-        `--${getEscapedCssVarName(key, false)}`,
-        String(cssVars[key]),
-      )
+      const value = cssVars[key]
+      if (isString(value) || typeof value === 'number') {
```

Comment:
should also take into account an empty string
see [Playground with this PR](https://deploy-preview-12442--vue-sfc-playground.netlify.app/#__SSR__eNp9kU9PAjEQxb/KpB4WEv4lelqBRAkxelCjHntZusO60G2btouYzX53p11AYgy39s2bN79pG3ZnzGhXI0vZ1GNlZOZxzhXA1ICQmXMzzlZSiy1nkDr/LZGEBoSW2qaQXE0mkwRazmIPgEfnY/fYkDIdn0XS1QlbGg8OfW1AZqqgLO9ic1kZbT00YHENLaytriAhriQ0Cq2ch1Umtg9W1ypfhOkwC95ekvTDoC76MCZgghPaYE7KKPJDE7hCRhEzhocVdsNVqfJe8ic96d9y1cbkkDZnAyIljnVZjDZOK3qvGMiZ0JUpJdoX40vi5CztRoVaJqX+eoqatzUOjrr4RLH9R9+4fdA4e7Xo0O6Qs1PNZ7ZA35WX78+4p/OpWOm8luS+UHxDp2UdGDvbPa1K2Ge+SPsYP6JUxYdb7j0qd1wqgAZnG/2c0ecsLqz+i3s9uol99J6s/QGejM/7)

Reply:
Do we even need to take care of values apart from `null` and `undefined` here? Even a string can cause mismatch here. eg. `";"`